packages/arb-node-core/challenge/generateProofCases.go

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2019-2020, Offchain Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package challenge
+
+import (
+	"fmt"
+	"math/big"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/pkg/errors"
+
+	"github.com/offchainlabs/arbitrum/packages/arb-avm-cpp/cmachine"
+	"github.com/offchainlabs/arbitrum/packages/arb-util/core"
+	"github.com/offchainlabs/arbitrum/packages/arb-util/inbox"
+	"github.com/offchainlabs/arbitrum/packages/arb-util/machine"
+)
+
+type ExecutionCutJSON struct {
+	GasUsed      uint64
+	InboxDelta   ethcommon.Hash
+	MachineState ethcommon.Hash
+	SendAcc      ethcommon.Hash
+	SendCount    *hexutil.Big
+	LogAcc       ethcommon.Hash
+	LogCount     *hexutil.Big
+}
+
+type ProofData struct {
+	BeforeCut   ExecutionCutJSON
+	AfterCut    ExecutionCutJSON
+	Proof       hexutil.Bytes
+	BufferProof hexutil.Bytes
+}
+
+func GenerateProofCases(contract string, fuzzTarget bool) ([]*ProofData, error) {
+	mach, err := cmachine.New(contract)
+	if err != nil {
+		if fuzzTarget {
+			return make([]*ProofData, 0), nil
+		}
+		return nil, err
+	}
+
+	maxSteps := uint64(100000)
+	if fuzzTarget {
+		maxSteps = 1000
+	}
+	db := core.NewValidatorLookupMock(mach)
+	for i := 0; i < 100; i++ {
+		db.AddMessage(inbox.NewRandomInboxMessage())
+	}
+
+	beforeCut := ExecutionCutJSON{
+		GasUsed:      0,
+		InboxDelta:   ethcommon.Hash{},
+		MachineState: mach.Hash().ToEthHash(),
+		SendAcc:      ethcommon.Hash{},
+		SendCount:    (*hexutil.Big)(big.NewInt(0)),
+		LogAcc:       ethcommon.Hash{},
+		LogCount:     (*hexutil.Big)(big.NewInt(0)),
+	}
+	nextMessageIndex := big.NewInt(0)
+	proofs := make([]*ProofData, 0)
+	for i := uint64(0); i < maxSteps; i++ {
+		proof, err := mach.MarshalForProof()
+		if fuzzTarget && len(proof) == 0 {
+			return proofs, nil
+		}
```

Comment:
I think the change to `MarshalForProof()` to return an error in the case of a zero length proof should eliminate the need for this special case